integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -1426,6 +1426,9 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
 
       await newRealm.subscriptions.update((mutableSubs) => subsUpdateFn(mutableSubs, newRealm));
 
+      // This was added to ensure all server changes are downloaded before any tests assertions
+      await newRealm?.syncSession?.downloadAllServerChanges();
```

Comment:
I'm surprised that this one is necessary as I thought `waitForSynchronization` (which `await .subscriptions.update` calls) should ensure all the matching objects are downloaded – or is that not the case?

Reply:
This is actually not needed 👍🏼